checker/src/org/checkerframework/checker/linear/LinearVisitor.java

```diff
@@ -69,6 +75,21 @@ public Void visitMemberSelect(MemberSelectTree node, Void p) {
         return super.visitMemberSelect(node, p);
     }
 
+    @Override
+    protected void commonAssignmentCheck(
+            AnnotatedTypeMirror varType,
+            AnnotatedTypeMirror valueType,
+            Tree valueTree,
+            String errorKey) {
+        if (varType.hasAnnotation(Linear.class)) {
+            if (valueTree instanceof LiteralTree || valueTree instanceof NewClassTree) {
```

Comment:
Use the org.checkerframework.framework.qual.ImplicitFor meta-annotation to specify that the type of all literals is `@Linear`.

Reply:
Is that better than this? Why? Seems good to be able to  put all cases together... or is there an annotation for `@ImplicitFor(NewClassTree)` as well?

Reply:
Also, not all literals are `@Linear`, only those assigned to a `@Linear` variable.

Reply:
Oh, so here you were actually trying to allow literals and NewClassTrees to be assigned to any `@Linear` reference rather than changing the types of those trees.  In that case, it would be better to just return true.

I thought you were trying to change the type of all literals to `@Linear`.

Reply:
Renado, if you could add comments to the code, that would help a lot in understanding and reviewing it -- we could give you better advice, more quickly.  Thanks!

Reply:
Return true where? I tried overriding `isAssignable` instead and return true in the cases above, but `isAssignable` was not even called?!